aodndata/moorings/products_handler.py

```diff
@@ -274,6 +275,39 @@ def _make_hourly_timeseries(self):
 
             self._cleanup_previous_version(product_file.name)
 
+    def _make_gridded_timeseries(self):
+        """Generage TEMP gridded product from the new hourly product file."""
+
+        # Find the full QC'd hourly product to use as input
+        hourly_files = (self.file_collection
+                            .filter_by_attribute_regex('name', r'.*FV02_hourly-timeseries_END')
+                            .filter_by_attribute_id('publish_type', PipelineFilePublishType.HARVEST_UPLOAD)
+                        )
+        if len(hourly_files) != 1:
+            raise MissingFileError("Don't have an hourly-timeseries file as input for gridded product!")
+        hourly_file = hourly_files[0]
+        self.logger.info("Creating gridded product from '{hourly_file.name}'".format(hourly_file=hourly_file))
+
+        # create a local symlink in temp_dir which includes the final S3 id (dest_path) so that
+        # it is recorded in the gridded file's metadata
+        hourly_file.dest_path = self.dest_path(hourly_file.local_path)
+        hourly_temp_path = os.path.join(self.temp_dir, hourly_file.dest_path)
+        os.makedirs(os.path.dirname(hourly_temp_path))
+        os.symlink(hourly_file.local_path, hourly_temp_path)
+
```

Comment:
is this shell-like hack really necessary !? I can fanthom why only with this codebase - apparently a plumbing to an aodncore missing functionality.

Reply:
aodncore has nothing to do with *how* NetCDF files are generated. 

This appears to be parameterising this function in aodntools according to the expectations of *that* function:

https://github.com/aodn/python-aodntools/blob/master/aodntools/timeseries_products/gridded_timeseries.py#L139

Reply:
That's right. As stated in the comment above (line 291), this is done so that the gridded product can include its source file's S3 id (making it directly accessible).

The alternative would be to add a separate argument to the gridding function, specifying this id. This would be ok in this case, but more complicated for the other products, which have a list of input files (often hundreds...), so ensuring the two lists are in sync would be a mess. So I'd rather just keep this solution.